test/e2e/specs/preview.test.js

```diff
@@ -148,19 +138,20 @@ describe( 'Preview', () => {
 		await editorPage.bringToFront();
 		await publishPost();
 		await Promise.all( [
-			page.waitForFunction( () => ! document.querySelector( '.editor-post-preview' ) ),
-			page.click( '.editor-post-publish-panel__header button' ),
+			editorPage.waitForFunction( () => ! document.querySelector( '.editor-post-preview' ) ),
+			editorPage.click( '.editor-post-publish-panel__header button' ),
 		] );
 		expectedPreviewURL = await editorPage.$eval( '.notice-success a', ( node ) => node.href );
-		previewPage = await getOpenedPreviewPage();
+		previewPage = await getOpenedPreviewPage( editorPage, previewPage );
 		expect( previewPage.url() ).toBe( expectedPreviewURL );
 
 		// Return to editor to change title.
 		await editorPage.bringToFront();
 		await editorPage.type( '.editor-post-title__input', ' And more.' );
 
 		// Published preview should reuse same popup frame.
-		previewPage = await getOpenedPreviewPage();
+		// TODO: Fix code to reuse the same frame!
+		previewPage = await getOpenedPreviewPage( editorPage );
```

Comment:
This is the only place where we don't reuse the same frame. Very surprising given given that there is code comment which states it 😃